src/main/java/frc/robot/systems/DriveFSMSystem.java

```diff
@@ -0,0 +1,693 @@
+
+		double xDiff = x - getPose().getX();
+		double yDiff = y - getPose().getY();
+		double aDiff = angle - getPose().getRotation().getDegrees();
+
+		SmartDashboard.putNumber("x diff", xDiff);
+		SmartDashboard.putNumber("y diff", yDiff);
+		SmartDashboard.putNumber("a diff", aDiff);
+
+		double xSpeed;
+		double ySpeed;
+		if (Math.abs(xDiff) > Math.abs(yDiff)) {
+			xSpeed = clamp(xDiff / AutoConstants.AUTO_DRIVE_TRANSLATIONAL_SPEED_ACCEL_CONSTANT,
+			-AutoConstants.MAX_SPEED_METERS_PER_SECOND, AutoConstants.MAX_SPEED_METERS_PER_SECOND);
+			ySpeed = xSpeed * (yDiff / xDiff);
+			if (Math.abs(xDiff) < AutoConstants.CONSTANT_SPEED_THRESHOLD && Math.abs(yDiff)
+				< AutoConstants.CONSTANT_SPEED_THRESHOLD) {
+				xSpeed = (AutoConstants.CONSTANT_SPEED_THRESHOLD * xDiff / Math.abs(xDiff))
+					/ AutoConstants.AUTO_DRIVE_TRANSLATIONAL_SPEED_ACCEL_CONSTANT;
+				ySpeed = xSpeed * (yDiff / xDiff);
+			}
+		} else {
+			ySpeed = clamp(yDiff / AutoConstants.AUTO_DRIVE_TRANSLATIONAL_SPEED_ACCEL_CONSTANT,
+			-AutoConstants.MAX_SPEED_METERS_PER_SECOND, AutoConstants.MAX_SPEED_METERS_PER_SECOND);
+			xSpeed = ySpeed * (xDiff / yDiff);
+			if (Math.abs(xDiff) < AutoConstants.CONSTANT_SPEED_THRESHOLD && Math.abs(yDiff)
+				< AutoConstants.CONSTANT_SPEED_THRESHOLD) {
+				ySpeed = (AutoConstants.CONSTANT_SPEED_THRESHOLD * yDiff / Math.abs(yDiff))
+					/ AutoConstants.AUTO_DRIVE_TRANSLATIONAL_SPEED_ACCEL_CONSTANT;
+				xSpeed = ySpeed * (xDiff / yDiff);
+			}
+		}
+
+		xSpeed = Math.abs(xDiff) > AutoConstants.AUTO_DRIVE_METERS_MARGIN_OF_ERROR
+			? xSpeed : 0;
+		ySpeed = Math.abs(yDiff) > AutoConstants.AUTO_DRIVE_METERS_MARGIN_OF_ERROR
+			? ySpeed : 0;
+		double aSpeed = Math.abs(aDiff) > AutoConstants.AUTO_DRIVE_DEGREES_MARGIN_OF_ERROR
+			? (aDiff > 0 ? Math.min(AutoConstants.MAX_ANGULAR_SPEED_RADIANS_PER_SECOND, aDiff
+			/ AutoConstants.AUTO_DRIVE_ANGULAR_SPEED_ACCEL_CONSTANT) : Math.max(
+			-AutoConstants.MAX_ANGULAR_SPEED_RADIANS_PER_SECOND, aDiff
+			/ AutoConstants.AUTO_DRIVE_ANGULAR_SPEED_ACCEL_CONSTANT)) : 0;
+
+		drive(xSpeed, ySpeed, aSpeed, true, false);
+		if (xSpeed == 0 && ySpeed == 0 && aSpeed == 0) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Drives the robot through a series of points.
+	 * @param points arraylist of points to drive to
+	 * @return if the robot has driven to the next position
+	 */
+	public boolean driveAlongPath(ArrayList<Pose2d> points) {
+		if (currentPointInPath >= points.size()) {
+			drive(0, 0, 0, true, false);
+			return true;
+		}
+		if (driveToPose(points.get(currentPointInPath))) {
+			currentPointInPath++;
+		}
+		return false;
+	}
+
+	/**
+	 * Drives the robot until it reaches a given position relative to the apriltag.
+	 * @param xDist constantly updating x distance to the point
+	 * @param yDist constantly updating y distance to the point
+	 * @param rotFinal constantly updating angular difference to the point
+	 */
+	public void driveToTag(double xDist, double yDist, double rotFinal) {
```

Comment:
Delete this method for now, will add back in the automation PR

Reply:
Thats so sad, it will be much easier to merge automation speaker alignment code into this pr bc they're already pretty close. I've been merging automation updates into drive.